components/Agents/AgentEditDialog.tsx

```diff
@@ -0,0 +1,98 @@
+import type React from "react";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { Button } from "@/components/ui/button";
+import { Pencil } from "lucide-react";
+import CreateAgentForm from "./CreateAgentForm";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { useUserProvider } from "@/providers/UserProvder";
+import type { AgentTemplateRow } from "@/types/AgentTemplates";
+import { useState } from "react";
+
+interface AgentEditDialogProps {
+  agent: AgentTemplateRow;
+}
+
+const AgentEditDialog: React.FC<AgentEditDialogProps> = ({ agent }) => {
+  const [open, setOpen] = useState(false);
+  const { userData } = useUserProvider();
+  const queryClient = useQueryClient();
+
+  const editTemplate = useMutation({
+    mutationFn: async (values: {
+      title?: string;
+      description?: string;
+      prompt?: string;
+      tags?: string[];
+      isPrivate?: boolean;
+    }) => {
+      const res = await fetch("/api/agent-templates", {
+        method: "PATCH",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          id: agent.id,
+          userId: userData?.id,
+          title: values.title,
+          description: values.description,
+          prompt: values.prompt,
+          tags: values.tags,
+          isPrivate: values.isPrivate,
+        }),
+      });
+      if (!res.ok) throw new Error("Failed to update template");
+      return res.json();
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["agent-templates"] });
+      setOpen(false);
+    },
+  });
```

Comment:
Single Responsibility Principle
- actual: React Query hook initialized within a react component.
- required: new hook created for `useEditAgentTemplate`